numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -38,6 +38,40 @@
 #include "npy_cblas.h"
 #include "npy_buffer.h"
 
+
+/*
+ * Define array fields with only the flags set. The flags are set to preserve
+ * alignment information in the unlikely case where they may be used,
+ * the only other field that should typically be set is the descriptor.
+ * (It is not set here, since sometimes it is set in a loop.)
+ *
+ * This object fakes a reference count of 1 (which should not really change,
+ * but allows it to be INCREF'd/DECREF'd if necessary (occurs during base
+ * setting at the time of writing this).
+ * It also sets a the base and the type. All of these are necessary because
+ * for subarrays a new array is created which requires this object to
+ * set the correct base. During base setting, it will be skipped so that
+ * we point to the original array object (or its base).
+ */
+#define _MAKE_DUMMY_STACK_ARRAY(arr, dummy_array)                   \
+        PyArrayObject_fields __dummy_array_fields;                  \
+        __dummy_array_fields.flags = PyArray_FLAGS(arr);            \
+        __dummy_array_fields.flags &= ~NPY_ARRAY_OWNDATA;           \
+        Py_TYPE(&__dummy_array_fields) = &PyArray_Type;             \
+        Py_REFCNT(&__dummy_array_fields) = 1;                       \
+        __dummy_array_fields.base = (PyObject *)arr;                \
+        *(dummy_array) = (PyArrayObject *)&__dummy_array_fields
+
+#ifndef NDEBUG
+    /* Fill data with bad value to error if free'd */
+    #define MAKE_DUMMY_STACK_ARRAY(arr, dummy_array)                \
+            _MAKE_DUMMY_STACK_ARRAY(arr, dummy_array);              \
+            __dummy_array_fields.data = 0xDEADBEEF
```

Comment:
Makes the tests reliably segfault in debug mode (before seems it was limited to windows when I forgot the removal of owndata).